includes/wc-template-functions.php

```diff
@@ -836,6 +836,10 @@ function woocommerce_pagination() {
 	 * @subpackage	Product
 	 */
 	function woocommerce_show_product_images() {
+		if ( ! $GLOBALS['product'] ) {
```

Comment:
Sorry, could actually use is_a here too since $product could be set to something invalid? Would protect us?